xlators/features/shard/src/shard.c

```diff
@@ -1064,23 +1064,26 @@ shard_common_resolve_shards(call_frame_t *frame, xlator_t *this,
     else
         gf_uuid_copy(gfid, local->base_gfid);
 
+    /* Build base shard path before appending index of the shard */
+    strcpy(path, "/" GF_SHARD_DIR "/");
+    uuid_utoa_r(gfid, path + sizeof(GF_SHARD_DIR) + 1);
+    int prefix_len = sizeof(GF_SHARD_DIR) + GF_UUID_BUF_SIZE;
+
```

Comment:
This is used several times. Could you create a function to initialize this ?

Reply:
Yes. Making it a macro